frontend/source/class/agrammon/module/output/SubmitWindow.js

```diff
@@ -261,42 +260,54 @@ qx.Class.define('agrammon.module.output.SubmitWindow', {
-            var lang         = 'lang='    + locale.replace(/_.+/,'');
-            var reportUrl    = 'reports=' + this.__reportSelected;
-            var titleUrl     = 'titles='  + this.__titleSelected;
-            var pidUrl       = 'pid='     + this.__outputData.getPid();
-            var sessionUrl   = 'session=' + this.__outputData.getSession();
-            var userUrl      = 'user='    + info.getUserName();
-            var datasetUrl   = 'dataset=' + info.getDatasetName();
-            var modelVariant = 'modelVariant='+agrammon.Info.getInstance().getModelVariant();
-            var guiVariant   = 'guiVariant='+agrammon.Info.getInstance().getGuiVariant();
-            var url = 'export/latex'
-                               + '?' + 'mode=kantonLU'
-                               + '&' + 'format=pdf'
-                               + '&' + reportUrl
-                               + '&' + titleUrl
-                               + '&' + pidUrl
-                               + '&' + sessionUrl
-                               + '&' + userUrl
-                               + '&' + datasetUrl
-                               + '&' + lang
-                               + '&' + guiVariant
-                               + '&' + modelVariant
-                               + '&' + 'version='   + this.__version
-                               + '&' + 'farm='      + this.__farmNumberInput.getValue()
-                               + '&' + 'sender='    + sender
-                               + '&' + 'comment='   + comment
-                               + '&' + 'recipient=' + this.__recipientSelect.getSelection()[0].getLabel()
-                               + '&' + 'email='     + this.__recipientSelect.getSelection()[0].getModel()
-                               + '&' + 'situation=' + this.__farmSituationSelect.getSelection()[0].getLabel()
-                               + '&' + 'log='       + this.__logMsg;
+            var lang         = locale.replace(/_.+/,'');
+
+            var params = {
+                language     : lang,
+                username     : info.getUserName(),
+                reports      : this.__reportSelected,
+                format       : 'excel',
+                titles       : this.__titleSelected,
+                datasetName  : info.getDatasetName(),
+                modelVariant : agrammon.Info.getInstance().getModelVariant(),
+                guiVariant   : agrammon.Info.getInstance().getGuiVariant(),
+                version      : this.__version,
+                mode         : 'submission',
+
+                farmSituation  : this.__farmSituationSelect.getSelection()[0].getLabel(),
+                farmNumber     : this.__farmNumberInput.getValue(),
+                comment        : comment,
+                senderName     : sender,
+                recipientName  : this.__recipientSelect.getSelection()[0].getLabel(),
+                recipientEmail : this.__recipientSelect.getSelection()[0].getModel()
```

Comment:
Seems we're exposing an email sending endpoint here that could potentially be abused to send spam? Or do we validate the email addresses on the server side too?

Reply:
I will move the recipient address(es) to the backend config, just wanted to make the current setting work first.

Reply:
I add a check for the recipient for the moment and redirect any mail not sent to me, to me for the moment.
Will have to think about this a bit. Will probably configure the allowed addresses for submission in the config file.

There is another mail functionality for "sending" datasets to other users (just a DB copy) with an eMail notification.
There will be a check for the recipient being a valid Agrammon user. Not sure I can make that completely save against Agrammon users spamming eachother.

Reply:
I think Agrammon users spamming each other is a much smaller (and ignorable) problem.

Reply:
> I think Agrammon users spamming each other is a much smaller (and ignorable) problem.

Agreed.